internal/cloudsql/refresh.go

```diff
@@ -141,16 +145,16 @@ func createTLSConfig(inst connName, m metadata, cert tls.Certificate) *tls.Confi
 		// certificates, we instead need to implement our own VerifyPeerCertificate function
 		// that will verify that the certificate is OK.
 		InsecureSkipVerify:    true,
-		VerifyPeerCertificate: genVerifyPeerCertificateFunc(inst, certs),
+		VerifyPeerCertificate: verifyPeerCertFunc(inst, certs),
 	}
 	return cfg
 }
 
-// genVerifyPeerCertificateFunc creates a VerifyPeerCertificate func that verifies that the peer
-// certificate is in the cert pool. We need to define our own because of our sketchy non-standard
-// CNs.
-func genVerifyPeerCertificateFunc(cn connName, pool *x509.CertPool) func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
-	return func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
+// verifyPeerCertFunc creates a VerifyPeerCertificate func that verifies that the peer
+// certificate is in the cert pool. We need to define our own because CloudSQL
+// instances use self-signed certificates as a RootCA.
```

Comment:
We need the verify function not because it's a self-signed cert, but instead because we use the instance name instead of a valid domain for the certificate name field.

Reply:
Fixed.